### PR TITLE
14.0 mig sale cancel reason

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

**Review by danimv5 — APPROVED** [review]
Functional review. LGTM!!

File: sale_cancel_reason/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright 2013 Guewen Baconnier, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import wizard
+from . import model
```
Comment (CasVissers-360ERP) [inline sale_cancel_reason/__init__.py:5]: Should be models?

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Minor feedback

File: sale_cancel_reason/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_sale_order_cancel_reason_user,access_sale_order_cancel_reason user,model_sale_order_cancel_reason,sales_team.group_sale_salesman,1,0,0,0
+access_sale_order_cancel_reason_manager,access_sale_order_cancel_reason manager,model_sale_order_cancel_reason,sales_team.group_sale_manager,1,1,1,1
```
Comment (atchuthan) [inline sale_cancel_reason/security/ir.model.access.csv:3]: missing security for wizard i.e. sale.order.cancel
In v14, we need to manage access rights for TransientModels too

**Review by atchuthan — CHANGES_REQUESTED** [review]

File: sale_cancel_reason/wizard/cancel_reason.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# © 2013 Guewen Baconnier, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class SaleOrderCancel(models.TransientModel):
+
+    """ Ask a reason for the sale order cancellation."""
+
+    _name = "sale.order.cancel"
```
Comment (CasVissers-360ERP) [inline sale_cancel_reason/wizard/cancel_reason.py:10]: Name already in use by base Odoo

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Incorrect model name and access rights are missing.

File: sale_cancel_reason/model/sale.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        "sale.order.cancel.reason",
+        string="Reason for cancellation",
+        readonly=True,
+        ondelete="restrict",
+        tracking=True,
+    )
+
+
+class SaleOrderCancelReason(models.Model):
+    _name = "sale.order.cancel.reason"
```
Comment (rousseldenis) [inline sale_cancel_reason/model/sale.py:19]: Time to move in separate file.

File: sale_cancel_reason/wizard/cancel_reason.py (hunk 3 above)
Comment (rousseldenis) [reply sale_cancel_reason/wizard/cancel_reason.py:10]: If you change, don't forget migration script.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by H74-code** [comment]
Hello, 

I'm really newbie in programming, but I would like to thank you guys for the effort of upgrading this module.

I'm looking forward towards using it.

Thanks once again!

**Comment by hparfr** [comment]
Migration done in https://github.com/OCA/sale-workflow/pull/2158

This PR should be closed

**Pull request closed without merging**